### ISSUE TITLE
Argument 2 passed to OpenApi\Analysis::addAnnotations() must be an instance of OpenApi\Context

### ISSUE THREAD

**Issue by jochen-jung** [zircote/swagger-php#997]
Since upgrading to 3.3.0 yesterday, I get the following error when creating the documentation:

`In Analysis.php line 128: Argument 2 passed to OpenApi\Analysis::addAnnotations() must be an instance of OpenApi\Context, null given, called in .../vendor/zircote/swagger-php/src/Analysis.php on line 85`

Works fine after downgrading again:
`Downgrading zircote/swagger-php (3.3.0 => 3.2.3)`

**Comment by DerManoMann** [comment]
Hmm, are you using swagger-php standalone or as part of a bundle or plugin?

**Comment by jochen-jung** [comment]
We have an own bundle which we are using as a wrapper around zircote/swagger-php

**Comment by phillylovepark** [comment]
Same here, using bundle `nelmio/api-doc-bundle` https://github.com/nelmio/NelmioApiDocBundle

**Comment by DerManoMann** [comment]
@phillylovepark Just checked that the NelmioApiDocBundle tests pass with the new release. Are you working on a public repo that I can access for testing?

@jochen-jung The code has changed the main contract is now to use the Generator class.  If you use any of the other classes directly you will need to update your code. Hard to judge the exact cause without looking at some actual code, unfortunately.

**Comment by jochen-jung** [comment]
Thanks for having a look, we will check that we always use the Generator class

**Comment by DerManoMann** [comment]
> Thanks for having a look, we will check that we always use the Generator class

👍 V4 will be out in a few weeks for PHP 8.1 and that has some more helpers for custom usage.

**Comment by DerManoMann** [comment]
@phillylovepark Just looking at this again - are you sure you are using the latest nelmio bundle version?
If so, could I ask you to recreate the issue and provide a full stacktrace? Looking at the bundle code I cannot see where/how that would have happened?

**Comment by phillylovepark** [comment]
I have just retried to upgrade project dependencies, with nelmio bundle v4.1.0 and swagger-php v3.3.1 there is no issue.
Last time I upgraded I have got same version of nelmio but with v3.3.0 of swagger-php.

If you want to, I could provide a reproduction project. Please tell me.

**Comment by DerManoMann** [comment]
A reproduction repo would be great as I couldn't break it by just running the nelmio tests

**Comment by phillylovepark** [comment]
Here it is : https://github.com/phillylovepark/symfony-skeleton/tree/swagger_997 

I started from Symfony skeleton, installed `nelmio/api-doc-bundle` and fixed `zircote/swagger-php` to v3.3.0.
Then you navigate to url `/api/doc.json` to trigger the error.

If you upgrade `zircote/swagger-php` to v3.3.1, then there is no error.

**Comment by DerManoMann** [comment]
Yep, can see that, so again, thanks for the repo.

I did some more digging which I should perhaps have done before... Your repo uses nelmio [v4.1.0](https://github.com/nelmio/NelmioApiDocBundle/releases/tag/v4.1.0), the changes to make the bundle work with the changes in swagger-php were added in [v4.4.0](https://github.com/nelmio/NelmioApiDocBundle/releases/tag/v4.4.0).

Looks like this is somewhat fixed due to the restriction on symfony 5.4.* (my guess). v4.1.0 is from December 2020...

Oh, well, either way - the context will be required in v4 of swagger-php :)
